src/pipeline_blocks.py

```diff
@@ -603,6 +608,43 @@ def _pos_cash_balance(config, train_mode, suffix, **kwargs):
         return pos_cash_balance
 
 
+def _previous_application_cleaning(config, suffix, **kwargs):
+    bureau_cleaning = Step(name='previous_application_cleaning{}'.format(suffix),
+                           transformer=dc.PreviousApplicationCleaning(**config.preprocessing.impute_missing),
+                           input_data=['previous_application'],
+                           adapter=Adapter({'previous_application': E('previous_application', 'X')}),
+                           experiment_directory=config.pipeline.experiment_directory,
+                           **kwargs)
+
+    return bureau_cleaning
```

Comment:
@pknut the naming should be previous_application_cleaning rather than bureau_cleaning I think